packages/api/src/useQuery.ts

```diff
@@ -15,9 +15,12 @@ const useQuery = <T extends TSocketEndpointNames>(name: T, ...props: TSocketAcce
     const prop = props?.[0];
     const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestPayload<T>) : undefined;
     const options = prop && 'options' in prop ? (prop.options as TSocketRequestQueryOptions<T>) : undefined;
-    const { send } = useAPI();
+    const { send, isReady } = useAPI();
 
-    return _useQuery<TSocketResponseData<T>, unknown>(getQueryKeys(name, payload), () => send(name, payload), options);
+    return _useQuery<TSocketResponseData<T>, unknown>(getQueryKeys(name, payload), () => send(name, payload), {
```

Comment:
I'm not sure we should be hiding/changing the `useQuery` method. Does it really follow the `SOLID` principles?

Reply:
@markwylde-deriv Yes, in my opinion it fixes logout. The main problem is useQuery doesn't know anything about WS connection but uses send function from there. In the legacy code (socket_base) we checked WS open or not before making any API request, but with new approach we don't do this. In result when we switches between real/demo and new WS connection opens useQuery made API requests when WS was not ready. I understand what you mean about SOLID, I can create another function useQueryWithWSCheck for example and change all needed hooks with it, but I'm pretty sure we needed to handle this behaviour in API layer earlier and it's not about open for extending/closed for modification, it's about bug fix. It's just my view :) Let me know your opinion and thanks a lot for your notes